### ISSUE TITLE
Velocity Waves/Foam

### ISSUE THREAD

**Issue by SaltSouls** [Goby56/wakes#48]
So, while I like the introduction of the waves and the foam particles(I do believe they could use a little work), the waves behavior is a bit wonky. For one, the way they look in a boat is really strange. You would imaging they would be more towards the sides of the boat, but currently they are clipping into the boat itself. It doesn't even look like they are coming from the water. Also, there seems to be an issue with their, and the wakes rendering, where they appear full bright even in dark areas.

As for the foam, I think this is an easy improvement. just make it less transparent, increase the particle quantity, make them stationary when they spawn, and increase their lifetime a little. I don't feel like they need to have any movement given that they should be the particles left behind as the entity moves through the water.

![2024-01-20_09 29 31](https://github.com/Goby56/wakes/assets/36361103/170fbe60-93c5-4680-bc9b-b8862c003ffa)
![2024-01-20_09 31 26](https://github.com/Goby56/wakes/assets/36361103/24aa52d5-3936-47f0-a558-7b3e10ed68ec)

PS. I should have also mentioned that the velocity waves lag behind the boat movement, so if you turn super fast, the waves will slowly move back towards the front of the boat instead of staying at the front.

**Comment by Goby56** [comment]
I fully agree on the particle part. They seem a bit out of place at the moment and I will definitely fix it some time in the future. For now though I need to address some bugs that just some people experience (and it seems as though you are one of them too). Wakes appear infront of some objects (the boat in your case), see issue #46. And the foamy white parts of the wakes take up too much space (it should look something like this): 
![preview-0](https://github.com/Goby56/wakes/assets/60710855/45378ea0-caa9-4d15-80a0-5209835334c5)
Then it also seems that the bug with fully lit wakes is back. 

Personally I have none of these issues with the latest version and I know that it also works for people I've recently played with. Just in case, make sure to fully reset your configs of the mod (i.e remove the whole config file "wakes.json" and let the mod create a new one). 

Because just some people experience similar issues I think the cause of the bugs lies with what hardware and possibly OS the user has. To make a complete picture of why these bugs may occur, could you provide what OS and graphics card you are using? In addition to this it would also be helpful if you cycled through the shader programs found in the debug settings and tell me if any of them makes it look better. (images are welcome)

**Comment by SaltSouls** [comment]
I'm running arch Linux on an AMD Radeon 6800xt. The last version of the mod didn't have any problems with rendering, aside from wakes getting darker when looked at from above.

As far as the shader that looks best, the general setting looks the best as it makes sure the wakes don't darken when looking at them from a top down view. Though as you can also see, it makes everything full bright, and makes the white sections take up way too much space. I will try deleting my config to see if that fixes any of the problems, and then get back to you.

**Comment by SaltSouls** [comment]
I can confirm that even after allowing the config to regenerate, I'm still getting these bugs.

**Comment by Goby56** [comment]
Is this with fancy or fabulous graphic settings? You see, I could only get wakes under water working with fabulous.

**Comment by Goby56** [comment]
The dark shading from above has been fixed with 0.2.3, see #34. Wakes under water is pretty complicated to fix due to how Minecraft's rendering engine works. For updates regarding below-water-visible-wakes see #92.
